onnxruntime/test/fuzzing/src/OnnxPrediction.cpp

```diff
@@ -2,40 +2,57 @@
 // Licensed under the MIT License.
 
 #include "OnnxPrediction.h"
+#include <vector>
+#include <memory>
 
 // Uses the onnxruntime to load the model
 // into a session.
 //
 OnnxPrediction::OnnxPrediction(std::wstring& onnx_model_file)
-:
-rawModel{nullptr},
-ptrSession{nullptr},
-session{ env, onnx_model_file.c_str(), emptySessionOption},
-inputNames{session.GetInputCount()},
-outputNames{session.GetOutputCount()}
-{
+    : rawModel{nullptr},
+      ptrSession{nullptr},
+      session{env, onnx_model_file.c_str(), emptySessionOption},
+      inputNames{session.GetInputCount()},
+      outputNames{session.GetOutputCount()} {
   init();
 }
 
 // Uses the onnx to seri
 //
 OnnxPrediction::OnnxPrediction(onnx::ModelProto& onnx_model)
-:
-session{nullptr}
-{
-  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),   
-  [this](void * ptr)
-      { 
-      this->GetAllocator().Free(ptr); 
-      }
-  };
+    : session{nullptr} {
+  rawModel = std::shared_ptr<void>{alloc.Alloc(onnx_model.ByteSizeLong()),
+                                   [this](void* ptr) {
+                                     this->GetAllocator().Free(ptr);
+                                   }};
 
   onnx_model.SerializeToArray(rawModel.get(), static_cast<int>(onnx_model.ByteSizeLong()));
 
-  ptrSession = std::make_unique<Ort::Session>(env, 
-      rawModel.get(), 
-      onnx_model.ByteSizeLong(), 
-      emptySessionOption),
+  ptrSession = std::make_unique<Ort::Session>(env,
+                                              rawModel.get(),
+                                              onnx_model.ByteSizeLong(),
+                                              emptySessionOption),
+
+  inputNames.resize(ptrSession->GetInputCount());
+  outputNames.resize(ptrSession->GetOutputCount());
+
+  init();
+}
+
+OnnxPrediction::OnnxPrediction(const std::vector<char>& modelData)
+    : session{nullptr} {
+  size_t numBytes = modelData.size();
+  rawModel = std::shared_ptr<void>{alloc.Alloc(numBytes),
+                                   [this](void* ptr) {
+                                     this->GetAllocator().Free(ptr);
+                                   }};
+  memcpy(rawModel.get(), modelData.data(), numBytes);
+  Ort::SessionOptions so;
```

Comment:
Do we need this complication? I assume the test runner will create a buffer that will remain valid whilst this instance is in use, or it can transfer ownership of that buffer here via a unique_ptr. Based on that I don't think we need to do another copy here, and even if we did I'm not aware of a reason why we'd need to complicate the allocation by using an ORT allocator.